### ISSUE TITLE
[Bug] [zos_copy] does not copy to data set if it does not exist

### ISSUE THREAD

**Issue by WladyslawBorysow** [ansible-collections/ibm_zos_core#301]
Hello once again,
I got into trouble with newest module zos_copy from release 1.4.0-beta.1
As far as I understand it is capable of copying into not existing ds (it can create it if it doesn't exist). In my case ds on mainframe is first created, exists for a few seconds and then is deleted and error is reported and copy of course fails.

![image](https://user-images.githubusercontent.com/68340319/135607310-f652ae4b-aeaa-4d65-ad26-1b6105c5df19.png)

I attach my files below:

[ansible.cfg.txt](https://github.com/ansible-collections/ibm_zos_core/files/7266492/ansible.cfg.txt)
[playbook.txt](https://github.com/ansible-collections/ibm_zos_core/files/7266530/playbook.txt)

For now I don't have any idea why is this hapening...
Thank You,
Wladyslaw

**Comment by adn1107** [comment]
I am not able to create this error. 

Have you tried a different HLQ for your dest dataset?

**Comment by WladyslawBorysow** [comment]
Hello,
I managed to catch screen when ds is being created (it disappears in a second)

**Comment by WladyslawBorysow** [comment]
Hello,
I managed to catch screen when ds is being created (it disappears in a second)
![image](https://user-images.githubusercontent.com/68340319/135814556-87e1e031-6d56-46c0-9ad4-9deee31e0fc4.png)
From the screen I conclude that there is some error with ds allocation. It is using 1 trk (primary,secondary) allocation and it is using all extends so it must fail. Seems it doesn't respect values provided by me. I changed TRK to M and had the same effect.

**Comment by adn1107** [comment]
Do you think SMS is interfering or something wrong with Ansible allocation of temp ds?

**Comment by ddimatos** [comment]
@WladyslawBorysow - can monitor progress [here](https://github.com/ansible-collections/ibm_zos_core/pull/306)
I have still have work to do on this but if you want to test drive the hotfix particular to the issue you reported, here is a binary you can install and try.
[ibm-ibm_zos_core-1.4.0-beta.2.tar.gz](https://github.com/ansible-collections/ibm_zos_core/files/7445439/ibm-ibm_zos_core-1.4.0-beta.2.tar.gz)

Use:
`ansible-galaxy collection install -f ibm-ibm_zos_core-1.*` from where this binary is located. You can later pull from Galaxy again with a -f (force) to erase this hotfix if you wish.

**Comment by ryanp54** [comment]
@ddimatos - I was having this same issue and that hotfix worked for me. The behavior I'm seeing does differ a little bit from the documentation though. As I read it I expect FB with LRECL of 80 as the default if I don't include 'destination_dataset', however I get VB with LRECL of 1028 if I don't specify anything.

**Comment by ddimatos** [comment]
Thank you @ryanp54 - I am still working on the next version of zos_copy and will take your comment into account.

**Comment by ddimatos** [comment]
@rexemin - you took over the work for `zos_copy` that is going into version 1.4.0-beta.2 scheduled for Nov 2022, the code currently in this branch [release-v1.4.0-beta.2](https://github.com/ansible-collections/ibm_zos_core/tree/release-v1.4.0-beta).
We could close this out based 2 things:
1) Validate that this works as it should because we call the underlying `ensure_present()` code
```
   - name: Copy first local binary file to ds
      zos_copy:
        src: /usr/bin/zipinfo
        dest: WWZ038.GSD##E1.FSA101.DUMPTERS.D030320.BIN
        destination_dataset:
          dd_type: BASIC
          space_primary: 27
          space_secondary: 10
          space_type: TRK
          record_format: FB
          record_length: 1024
          block_size: 6144
        is_binary: true
      register: result
```
2) We no longer rely on `cp` to @ryanp54 's point, "As I read it I expect FB with LRECL of 80 as the default if I don't include 'destination_dataset', however I get VB with LRECL of 1028 if I don't specify anything." could you double check that our docs are updated to reflect what we decided on that should be default behavior. I know we spent time discussing the default based on if its binary or text and will adjust the default when a data set is not defined, we just need to make sure its in the doc and its updated. 

If we can do these 2 things, we can close this git issue out.

**Comment by ddimatos** [comment]
Thanks @rexemin - ill close this as its fixed in 1.4.0-beta.2 coming out early October 2022, if not sooner. If it needs to be reopened on 1.4.0-beta.2 it may be, otherwise this is a `FIN` (Fixed in Next), so lets close it.
